### PR TITLE
xfpga: fix static code analysis issue

### PR DESCRIPTION
**Pull request by anandaravuri**

### CODE REVIEW

File: libopae/plugins/xfpga/sysfs.c [hunk 1]
```diff
@@ -266,13 +266,22 @@ STATIC void find_resources(sysfs_fpga_region *region)
...
+		if (reg_res) {
+			regerror(reg_res, &re, err, 128);
+			FPGA_MSG("Error compiling regex: %s", err);
+		}
 	}
 
 	dir = opendir(region->region_path);
+	if (!dir) {
+		FPGA_MSG("failed to open region path: %s", region->region_path);
+		return;
```
Comment (tswhison) [inline libopae/plugins/xfpga/sysfs.c]: I think re is leaked here.

File: libopae/plugins/xfpga/sysfs.c [hunk 2]
```diff
@@ -266,13 +266,22 @@ STATIC void find_resources(sysfs_fpga_region *region)
...
-	if (reg_res) {
-		regerror(reg_res, &re, err, 128);
-		FPGA_MSG("Error compiling regex: %s", err);
+
+	if (SYSFS_FORMAT(sysfs_resource_fmt)) {
+
+		reg_res = regcomp(&re, SYSFS_FORMAT(sysfs_resource_fmt), REG_EXTENDED);
+		if (reg_res) {
+			regerror(reg_res, &re, err, 128);
+			FPGA_MSG("Error compiling regex: %s", err);
```
Comment (tswhison) [inline libopae/plugins/xfpga/sysfs.c]: Shouldn't we return an error code here, similar to 401?

File: libopae/plugins/xfpga/sysfs.c [hunk 3]
```diff
@@ -381,18 +390,32 @@ int sysfs_initialize(void)
...
+
+	if (SYSFS_FORMAT(sysfs_region_fmt)) {
+
+		reg_res = regcomp(&region_re, SYSFS_FORMAT(sysfs_region_fmt),
+			REG_EXTENDED);
+		if (reg_res) {
+			regerror(reg_res, &region_re, err, 128);
+			FPGA_ERR("Error compling regex: %s", err);
+			return FPGA_EXCEPTION;
+		};
```
Comment (tswhison) [inline libopae/plugins/xfpga/sysfs.c]: Remove extraneous semicolon.

File: libopae/plugins/xfpga/sysfs.c [hunk 4]
```diff
@@ -381,18 +390,32 @@ int sysfs_initialize(void)
...
+	}
 
 	const char *sysfs_class_fpga = SYSFS_FORMAT(sysfs_class_path);
+	if (!sysfs_class_fpga) {
+		FPGA_ERR("Invalid fpga class path: %s", sysfs_class_fpga);
+		return FPGA_EXCEPTION;
```
Comment (tswhison) [inline libopae/plugins/xfpga/sysfs.c]: I think region_re is leaked here.

File: libopae/plugins/xfpga/sysfs.c [hunk 5]
```diff
@@ -381,18 +390,32 @@ int sysfs_initialize(void)
...
+	}
+
 	// open the root sysfs class directory
 	// look in the directory and get region (device) objects
 	dir = opendir(sysfs_class_fpga);
+	if (!dir) {
+		FPGA_MSG("failed to open region path: %s", sysfs_class_fpga);
+		return FPGA_EXCEPTION;
```
Comment (tswhison) [inline libopae/plugins/xfpga/sysfs.c]: I think region_re is leaked here.

File: libopae/plugins/xfpga/sysfs.c (hunk 2 above)
Comment (anandaravuri) [reply libopae/plugins/xfpga/sysfs.c]: fixed

File: libopae/plugins/xfpga/sysfs.c (hunk 1 above)
Comment (anandaravuri) [reply libopae/plugins/xfpga/sysfs.c]: fixed

File: libopae/plugins/xfpga/sysfs.c (hunk 3 above)
Comment (anandaravuri) [reply libopae/plugins/xfpga/sysfs.c]: fixed

File: libopae/plugins/xfpga/sysfs.c (hunk 4 above)
Comment (anandaravuri) [reply libopae/plugins/xfpga/sysfs.c]: fixed

File: libopae/plugins/xfpga/sysfs.c (hunk 5 above)
Comment (anandaravuri) [reply libopae/plugins/xfpga/sysfs.c]: fixed

File: libopae/plugins/xfpga/sysfs.c [hunk 6]
```diff
@@ -381,18 +396,34 @@ int sysfs_initialize(void)
...
+			FPGA_ERR("Error compling regex: %s", err);
+			return FPGA_EXCEPTION;
+		}
+	}
 
 	const char *sysfs_class_fpga = SYSFS_FORMAT(sysfs_class_path);
+	if (!sysfs_class_fpga) {
+		FPGA_ERR("Invalid fpga class path: %s", sysfs_class_fpga);
+		res = FPGA_EXCEPTION;
+		goto out_free;
```
Comment (tswhison) [inline libopae/plugins/xfpga/sysfs.c]: dir hasn't been opened yet, but line 481 will attempt to close it.

File: libopae/plugins/xfpga/sysfs.c [hunk 7]
```diff
@@ -257,22 +257,34 @@ STATIC sysfs_fpga_resource *make_resource(sysfs_fpga_region *region, char *name,
...
+			FPGA_MSG("Error compiling regex: %s", err);
+			return FPGA_EXCEPTION;
+		}
 	}
 
 	dir = opendir(region->region_path);
+	if (!dir) {
+		FPGA_MSG("failed to open region path: %s", region->region_path);
+		res = FPGA_EXCEPTION;
+		goto out_free;
```
Comment (tswhison) [inline libopae/plugins/xfpga/sysfs.c]: Line 321 will attempt to close dir, which wasn't opened successfully.

File: libopae/plugins/xfpga/sysfs.c (hunk 6 above)
Comment (tswhison) [reply libopae/plugins/xfpga/sysfs.c]: I think this can be fixed by changing 479 to..
```c
if (dir)
        closedir(dir);
```

**Pull request merged**